### ISSUE TITLE
Change of AQI, IAI and Total Volatile Organic Compounds UNIT

### ISSUE THREAD

**Issue by Gh61** [kongo09/philips-airpurifier-coap#116]
Hello,

thank you for amazing work and fast implementation of new models.

However, I ran across the issue, that in my chart on dashboard for IAI and some other sensors, there is nothing rendered. I found that the statistics are empty. So I searched through the logs and found this kind of errors:

```
The unit of sensor.kdekoliv_air_quality_index (None) cannot be converted to the unit of previously compiled statistics (AQI). Generation of long term statistics will be suppressed unless the unit changes back to AQI or a compatible unit. Go to https://my.home-assistant.io/redirect/developer_statistics to fix this
The unit of sensor.kdekoliv_indoor_allergen_index (None) cannot be converted to the unit of previously compiled statistics (IAI). Generation of long term statistics will be suppressed unless the unit changes back to IAI or a compatible unit. Go to https://my.home-assistant.io/redirect/developer_statistics to fix this
The unit of sensor.kdekoliv_total_volatile_organic_compounds (µg/m³) cannot be converted to the unit of previously compiled statistics (Level). Generation of long term statistics will be suppressed unless the unit changes back to Level or a compatible unit. Go to https://my.home-assistant.io/redirect/developer_statistics to fix this
```

When I go the proposed URL, I find this issues:
![image](https://github.com/kongo09/philips-airpurifier-coap/assets/10837736/32e13509-1bc7-4624-a850-5821eb558602)

It looks like I can simply fix it by pressing the button at the end of the line, but I believe the previous units were correct.

I don't know exactly which version first created this issue, but I believe it was around 0.17.0
If you need to know exactly, I believe I can test older versions and see if the issue still persists.

I am running the AC3858/51 model.

Can you please look at this? Thank you

**Comment by kongo09** [comment]
You're right, I changed the units some time ago as I thought these were not real units (as in: units of the SI system, like those that are built-in and recognized by Home Assistant. I essentially invented the former units and hardcoded them with the sensors as strings.

Now with your issue, I'm not so sure anymore. Maybe we can collect some more opinions? It would be easy enough to change back, if that's the right thing to do.

**Comment by Gh61** [comment]
Ok, the AQI unit is correct:
![image](https://github.com/kongo09/philips-airpurifier-coap/assets/10837736/07dd57f6-015f-4196-a8c5-0feb5e4393b0)

The IAI is proprietary something invented by Philips, it probably also does not have unit.
We only know it's in the range 1 - 12.

The "Total Volatile Organic Compounds" sensor does have the value of "Gas level" (Gas) from my purifier inside.
Similarly it's some proprietary metric by Philips, it probably does not have unit.
We only know possible values are: Level 1 - Level 4

So the AQI and IAI is probably best to leave at None.
But the Total Volatile Organic Compounds should be also **changed to None**.

OR

The logical sensor name/identification is incorrectly paired with value on my AC3858/51, as this should be Gas Level sensor with None as unit.

**Comment by kongo09** [comment]
Thanks for researching this, very helpful!

So your proposal is to move TOC to unit `Level`? And leave the others at `None`?

**Comment by Gh61** [comment]
Yes, that would be perfect.

**Comment by kongo09** [comment]
As a unit, do you think `L` is better or the full word `Level`?

**Comment by Gh61** [comment]
I believe the full word might be better, as it is still pretty short.
But if you want it shorter, `LVL` seems to be preferred abbreviation.
https://www.abbreviations.com/abbreviation/Level

**Comment by kongo09** [comment]
Philips seems to use `L1` to `L4` so using `L` feels closer to that. Alternatively, I could convert the number to the string so that it looks like the Philips reading. But HA wouldn't understand that to be a quantity.

Overall, I guess using just `L` is most in line with the Philips way of doing it.

**Comment by Gh61** [comment]
You're right. Using number + `L` as unit would be closest to Philips og app.
Thank you

**Comment by kongo09** [comment]
fixed in 0.18.5-beta.1
